File: JavaOOPBasics/05.Inheritance-Exercises/src/P03_Mankind/Student.java
package P03_Mankind;

class Student extends Human {
    private String  facultyNumber;

    public Student(String studentName, String studentLastName, String facultyNumber) {
        super(studentName, studentLastName);
        setFacultyNumber(facultyNumber);
    }

    private String getFacultyNumber() {
        return facultyNumber;
    }

    private void setFacultyNumber(String  facultyNumber) {
        if (facultyNumber.length() < 5 || facultyNumber.length() > 10) {
            throw new IllegalArgumentException("Invalid faculty number!");
        } else {
            this.facultyNumber = facultyNumber;
        }
    }

    @Override
    public String toString() {
        return String.format("First Name: %s%nLast Name: %s%nFaculty number: %s%n",
                super.getFirstName(),
                super.getLastName(),
                this.getFacultyNumber()
        );
    }
}

File: Databases-Advanced-Entity-Framework/03.CodeFirst/01.OOPIntro/Student.cs
namespace _01.OOPIntro
{
    class Student
    {
        private string name;
        private static int counter;

        public static int Counter
        {
            get
            {
                return counter;
            }
        }

        public Student(string name)
        {
            this.name = name;
            ++counter;
        }

    }
}
